### PR TITLE
get services for each category from database

### LINKED ISSUE #40: Get api/v1/category/:id/services
- [ ] Route
 - [ ] Controller
 - [ ] Query

-----------------------------------------------------------------------
### Route
GET => api/v1/category/:id/services
apiRouter.get('/category/:id/services', getServices);

-----------------------------------------------------------------------
### Controller
const getServices =  () => {}

-----------------------------------------------------------------------
### Query
const getServices Query =  (id) => {}

### PR DESCRIPTION
**Pull request by Jehad91**
Relates #40
close #40 
image

### CODE REVIEW

File: server/routes/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { Router } from 'express';
-import { getCategories } from '../controllers';
+import { getCategories, getCategoryServices } from '../controllers';
 
 const router = Router();
 router.get('/categories', getCategories);
+router.get('/category/:id/services', getCategoryServices);
```
Comment (Mu7ammadAbed) [inline server/routes/index.ts:6]:
```suggestion
router.get('/categories/:id/services', getCategoryServices);
```

File: server/controllers/categories/getCategoryServices.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { RequestHandler } from 'express';
+import { getCategoryServicesQuery } from '../../queries';
+
+const getCategoryServices: RequestHandler = async (req, res, next) => {
+  const { id } = req.params;
```
Comment (Mu7ammadAbed) [inline server/controllers/categories/getCategoryServices.ts:5]: We need to validate this

File: server/controllers/categories/getCategoryServices.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  try {
+    const { rows } = await getCategoryServicesQuery(id as any);
+    res.json({ message: 'Successfully retrieved services', status: 200, data: rows });
```
Comment (Mu7ammadAbed) [inline server/controllers/categories/getCategoryServices.ts:9]:
```suggestion
    if (!(Number(id) > 0)) return res.status(400).json({ message: 'Category Id must be a number latger than 0' });
    const { rows, rowCount } = await getCategoryServicesQuery(id as any);
    if (!rowCount) return res.status(404).json({ message: 'Theres no category with this ID' });
    return res.json({ message: 'Successfully retrieved services', status: 200, data: rows });
```

**Review by Mu7ammadAbed — APPROVED** [review]

File: server/validation/categroyIdSchema.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import {
+  number, object,
+} from 'yup';
+
+const categroyIdSchema = object({
+  id: number().min(0).required('Category Id must be a number larger than 0'),
```
Comment (Mu7ammadAbed) [inline server/validation/categroyIdSchema.ts:6]:
```suggestion
  id: number().min(1).required('Category Id must be a number larger than 0'),
```

**Review by MohammadAlHabil — APPROVED** [review]

**Review by IsraaHamdi — APPROVED** [review]

**Review by MostafaQanoo — APPROVED** [review]

**Pull request merged**